spec/models/manageiq/providers/embedded_ansible/provider_spec.rb

```diff
@@ -66,4 +66,50 @@
     subject.url = "server.example.com"
     expect(subject.url).to eq("https://server.example.com/api/v1")
   end
+
+  context "DefaultAnsibleObjects concern" do
+    before do
+      subject.save
+    end
+
+    context "with no attributes" do
+      %w(organization credential inventory host).each do |obj_name|
+        it "#default_#{obj_name} returns nil" do
+          expect(subject.public_send("default_#{obj_name}")).to be_nil
+        end
+
+        it "#default_#{obj_name}= creates a new custom attribute" do
+          subject.public_send("default_#{obj_name}=", obj_name.length)
+          expect(subject.default_ansible_objects.find_by(:name => obj_name).value.to_i).to eq(obj_name.length)
+        end
+      end
+    end
+
+    context "with attributes saved" do
+      before do
+        %w(organization credential inventory host).each do |obj_name|
+          subject.default_ansible_objects.create(:name => obj_name, :value => obj_name.length)
+        end
+      end
+
+      %w(organization credential inventory host).each do |obj_name|
+        it "#default_#{obj_name} returns the saved value" do
+          expect(subject.public_send("default_#{obj_name}")).to eq(obj_name.length)
+        end
+
+        it "#default_#{obj_name}= raises an error if we try to set to a new value" do
+          expect { subject.public_send("default_#{obj_name}=", obj_name.length + 1) }.to raise_error(described_class::DefaultAnsibleObjectError)
+        end
+
+        it "#default_#{obj_name}= raises an error if we try to set to a new value" do
```

Comment:
Ah thanks I think I meant this to test that it worked if we tried to set the value to the same value ... Good catch!